app/templates/static/image_manipulation.js

```diff
@@ -4,10 +4,14 @@ async function downloadImage(imageElement, text, headlineElement, fontSize, colo
     const imageUrl = imageElement.dataset.blob || imageElement.src;
     const headline = headlineElement.textContent.trim();  // Get the headline text
 
+    // Sanitize the headline to create a valid filename
+    const sanitizedHeadline = headline.replace(/\s+/g, '_');
```

Comment:
What is the point of the sanitization done inside the `download-image` endpoint? Does that even matter?

Reply:
The sanitization inside the `download-image` endpoint ensures that the filename is valid and consistent when the image is processed and returned by the backend. However, for the frontend, we need to sanitize the headline before initiating the download to ensure the filename matches the expected format. Both sanitizations are necessary to maintain consistency and avoid any potential issues with invalid filenames.